icefoganalytics/travel-authorization
# Fatal if Condition in Edit My Travel Authorization Layout

Fixes https://github.com/icefoganalytics/travel-authorization/issues/112

Relates to:
- https://github.com/icefoganalytics/travel-authorization/pull/103 and specifically https://github.com/icefoganalytics/travel-authorization/commit/6c8f25df9006f90ab98a864d7c2007a16b8cb5f9

# Context

On the My Travel Requests edit page, when you click "Generate Estimates" all content in the "Details" tab gets wiped

**To Reproduce**
Steps to reproduce the behavior:
1. Go to the My Travel Request page via the top dropdown nav.
2. Click on the "+ Travel Authorization" button to create a new travel request
3. Scroll down to "Approvals" pane.
4. Click on the "Generate Estimates" button.
5. See the page content disappear.

**Screenshots**
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/e68a3137-1113-4a8f-a878-ce3472c9e1f4)

Page content missing
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/17b31714-9bec-4c9e-b069-39634f8e5f99)

**Additional context**

Via judicious use of `git bisect` I have determined that the bug was introduced in https://github.com/icefoganalytics/travel-authorization/pull/103 by https://github.com/icefoganalytics/travel-authorization/commit/6c8f25df9006f90ab98a864d7c2007a16b8cb5f9

# Implementation

Ensure that newly created Travel Authorizations have the minimum number of stops. Do this in the back-end rather than dynamically in the front-end. Courtesy of @anlek
Fix hacky code that was updating travel authorization expenses when navigating away from the "Estimate" tab. Now the Estimate tab updates the currently active travel authorization.

# Testing Instructions

1. Boot the app via `dev up`
2. Log in to the app at http://localhost:8080
3. Go to the My Travel Request page via the top dropdown nav.
4. Click on the "+ Travel Authorization" button to create a new travel request
5. Scroll down to "Approvals" pane.
6. Check that the default trip type is "Round Trip".
7. Check that the travel methods and types of accommodation are the expected default values.
8. Fill in the "Details" section.
9. Scroll to the bottom and click the "Save Draft" button.
10. Go back to the Details section and switch between every combination of trip types (round -> one way, one way -> round, round -> multi, multi -> round, one way -> multi, multi -> one way). Check that the defaults are what you would expect them to be.
11. Click on the "Generate Estimates" button.
12. See the page content _not_ disappear.
